src/containers/WelcomePage/WelcomePage.jsx

```diff
@@ -163,6 +255,86 @@ export class WelcomePage extends Component {
   }
 }
 
+function mapSearchToFrontPageStructure(data, t) {
+  if (
+    !data ||
+    (data && !data.search) ||
+    (data && data.search && !data.search.results) ||
+    (data &&
+      data.search &&
+      data.search.results &&
+      data.search.results.length === 0)
+  ) {
+    return [];
+  }
+  const result = data.search.results;
+  // grouping
+  const subjects = {
+    title: `${t('contentTypes.subject-material')}:`,
+    contentType: 'results-frontpage',
+    resources: [],
+  };
+  const topics = {
+    title: `${t('subjectPage.tabs.topics')}:`,
+    contentType: 'results-frontpage',
+    resources: [],
+  };
+  const resource = {
+    title: `${t('resource.label')}:`,
+    contentType: 'results-frontpage',
+    resources: [],
+  };
+  result.map(i => {
+    if (i && i.contexts && i.contexts.length) {
+      i.contexts.map(ctx => {
```

Comment:
forEach?

Reply:
"map" er kjappere enn "forEach" og lettere å lese (mindre bokstaver). Men i teorien så trenger vi ikke retur verdien her og den kunne ha blitt skrevet som forEach eller "for in"- loop ...